### PR TITLE
Issue 17 - Implement basic inventory

### PR DESCRIPTION
**Pull request by progier389**
Adding ds389_server action  plugin
A basic Inventory tested with playbooks that creates 4 docker containers and install an 389ds instance on each one

Issue: [17](https://github.com/389ds/ansible-ds/issues/17)

Reviewd by @droideck (Thanks!)

### CODE REVIEW

**Comment by progier389** [comment]
A first try which work quite well for basic configuration.
  I have create playbooks that
-  creates 4 VM 
- install 389ds on the vm
- install the ansible_ds module on the vm 
- configure a 389 ds instance with userroot backend

Doing that I learned a few things and some things that Rich has told us now makes sense:
  - We need an action plugin to read the inventory and generates the parameters for the module 
  - I understand now how we could have a single module (that is because playbook interact with different plugins)
  - I think that when the code will stabilize the module should be delivered with ds389 (maybe with lib389)
  - We could keep the module imterface as it is (or nearly) because the mapping is done at the plugin level
  - I tried an inventory with replicated topology, but so far I did not find a simple working schema
    (nested directory list makes things quite complex to overwrite a parameter in group variables by one in subgroup or in host
 - About replication as the inventory contains the full topology I think we should not strictly map to the replica and agreement from the inventory to 389ds one:
    - IMHO The replica should also have the agmt parameter related about how the agmt connects to the target host
     so the agmt will only needs to know which instance (or even host or group)  

Still working on trying to get a smart inventory model for replicated topology:

**Comment by progier389** [comment]
Ok, I rebased and added a replicated topology inventory  example with a scheme that should not be too cumbersome for users
I did not created yet the test playbook because the plugin and module does not yet support that inventory.
( I plan to do that in another ticket)

File: ansible_collections/ds389/ansible_ds/tests/inventory/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+| common                     | This directory contains common playbooks and utilities                                                 |
+| common/create_vms.yml      | A playbook that creates the ldapserver                                                                 |
+| common/create_vms.yml      | A playbook that creates docker containers for ldapservers hosts                                        |
+| common/destroy_vms.yml     | A playbook that removes docker containers for ldapservers hosts                                        |
+| common/init_vault.sh       | A script that generates the encrypted vault on all inventories                                         |
+| common/install_module.yaml | A playbook that creates install the ds389.ansible_ds module on ldapservers hosts                       |
+| common/run_playbook.sh     | A wrapper around ansible-playbook                                                                      |
+| common/run_playbook.sh     | A wrapper around ansible-inventory                                                                     |
```
Comment (droideck) [inline ansible_collections/ds389/ansible_ds/tests/inventory/README.md:31]: Same here. Is it a duplicate or was it intentional?

File: ansible_collections/ds389/ansible_ds/tests/inventory/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Files description
+
+| *File*                     | *Description*                                                                                          |
+| -------------------------- | ------------------------------------------------------------------------------------------------------ |
+| vault.clear                | The clear text vault file. In a production environment this file only exist encrupted in the inventory |
+| install_ds_on_vms.yaml     | Playbook that creates ds389 instance                                                                   |
+| vault.pw                   | The vault password file (should not exists in a production environment)                                |
+| common                     | This directory contains common playbooks and utilities                                                 |
+| common/create_vms.yml      | A playbook that creates the ldapserver                                                                 |
+| common/create_vms.yml      | A playbook that creates docker containers for ldapservers hosts                                        |
```
Comment (droideck) [inline ansible_collections/ds389/ansible_ds/tests/inventory/README.md:26]: Looks like a duplicate...

**Review by droideck — APPROVED** [review]
Besides the minor issue, looks great! Ack!

**Pull request merged**